scripts/prepare_sector_network.py

```diff
@@ -51,6 +51,33 @@ def add_lifetime_wind_solar(n):
         n.generators.loc[[index for index in n.generators.index.to_list()
                         if carrier in index], 'lifetime']=costs.at[carrier_name,'lifetime']
 
+
+def create_network_topology(n, prefix):
+    """
+    create a network topology as the electric network,
+    returns a pandas dataframe with bus0, bus1 and length
+    """
+
+    topo = pd.DataFrame(columns=["bus0", "bus1", "length"])
+    connector = " -> "
+    attrs = ["bus0", "bus1", "length"]
+
+    candidates = pd.concat([n.lines[attrs],
+                            n.links.loc[n.links.carrier == "DC", attrs]])
+
+    positive_order = candidates.bus0 < candidates.bus1
+    candidates_p = candidates[positive_order]
+    candidates_n = (candidates[~ positive_order]
+                    .rename(columns={"bus0": "bus1", "bus1": "bus0"}))
+    candidates = pd.concat((candidates_p, candidates_n), sort=False)
+
+    topo = candidates.groupby(["bus0", "bus1"], as_index=False).mean()
+    topo.rename(index=lambda x: prefix + topo.at[x, "bus0"]
+                + connector + topo.at[x, "bus1"],
+                inplace=True)
```

Comment:
Slightly more compact:
```suggestion
    topo.index = topo.apply(lambda x: prefix + x.bus0 + connector + x.bus1, axis=1)
```